### PR TITLE
Implement Kernel#remove_instance_variable

### PR DESCRIPTION
**Pull request by vais**

### CODE REVIEW

File: opal/corelib/kernel.rb [hunk 1]
```diff
@@ -787,6 +787,20 @@ def instance_variable_set(name, value)
     `self[Opal.ivar(name.substr(1))] = value`
   end
 
+  def remove_instance_variable(name)
+    name = Opal.instance_variable_name!(name)
+
+    %x{
+      var key = Opal.ivar(name.substr(1)),
+          val = self[key];
+      if (val === undefined) {
```
Comment (elia) [inline opal/corelib/kernel.rb]: Any reason for not checking with `hasOwnProperty` here?

Comment (vais) [inline opal/corelib/kernel.rb]: Good point!

Comment (IlyaBylich) [inline opal/corelib/kernel.rb]: @elia Does `Kernel#instance_variable_get` also need this check?

Comment (elia) [inline opal/corelib/kernel.rb]: @IlyaBylich I think it makes sense, unless we discover some weird behavior of ivars makes sense to ignore any possible property coming from the prototype.

**Pull request merged**